docs/apache-airflow/core-concepts/executor/index.rst

```diff
@@ -78,3 +71,134 @@ There are two types of executor - those that run tasks *locally* (inside the ``s
 .. note::
 
     New Airflow users may assume they need to run a separate executor process using one of the Local or Remote Executors. This is not correct. The executor logic runs *inside* the scheduler process, and will run the tasks locally or not depending the executor selected.
+
+Writing Your Own Executor
+-------------------------
+
+All Airflow executors implement a common interface so that they are pluggable and any executor has access to all abilities and integrations within Airflow. Primarily, the Airflow scheduler uses this interface to interact with the executor, but other components such as logging, CLI and backfill do as well.
+The public interface is the :class:`~airflow.executors.base_executor.BaseExecutor`. You can look through the code for the most detailed and up to date interface, but some important highlights are outlined below:
```

Comment:
Might just be me, but the colon at the end here feels awkward.  It is followed by a note block, then by another statement which ends in a colon.

Reply:
Yupp, fair point. I'll just terminate that in a period!